botocore/configloader.py

```diff
@@ -142,12 +143,13 @@ def raw_config_parse(config_filename, parse_subsections=True):
         path = os.path.expandvars(path)
         path = os.path.expanduser(path)
         if not os.path.isfile(path):
-            raise botocore.exceptions.ConfigNotFound(path=path)
-        cp = configparser.RawConfigParser()
+            raise botocore.exceptions.ConfigNotFound(path=_unicode_path(path))
+        cp = six.moves.configparser.RawConfigParser()
         try:
-            cp.read(path)
-        except configparser.Error:
-            raise botocore.exceptions.ConfigParseError(path=path)
+            cp.read([path])
```

Comment:
@JordonPhillips What's the rationale for this?

This is unrelated to your change which should just be decoding to unicode for config errors.

If this is because of some other bug let's address that as a separate PR with a separate set of tests.  I really just want to address the actual customer pain here which is masking errors due to not properly decoding issues.   This change is not directly related to that.